app/src/test/java/com/kickstarter/features/search/viewmodel/SearchAndFilterViewModelTest.kt

```diff
@@ -0,0 +1,84 @@
+package com.kickstarter.features.search.viewmodel
+
+import com.kickstarter.KSRobolectricTestCase
+import com.kickstarter.features.search.data.SearchEnvelope
+import com.kickstarter.mock.factories.ProjectFactory
+import com.kickstarter.mock.services.MockApolloClientV2
+import com.kickstarter.models.Project
+import com.kickstarter.services.DiscoveryParams
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.toList
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.test.UnconfinedTestDispatcher
+import kotlinx.coroutines.test.runTest
+import org.junit.Test
+
+@OptIn(ExperimentalCoroutinesApi::class)
+class SearchAndFilterViewModelTest : KSRobolectricTestCase() {
+
+    private lateinit var viewModel: SearchAndFilterViewModel
+
+    @Test
+    fun `test for initial state getPopularProjects will only have sorting parameter POPULAR`() = runTest {
+
+        var params: DiscoveryParams? = null
+        val projectList = listOf(ProjectFactory.project(), ProjectFactory.prelaunchProject(""))
+        val dispatcher = UnconfinedTestDispatcher(testScheduler)
+        val environment = environment()
+            .toBuilder()
+            .apolloClientV2(
+                object : MockApolloClientV2() {
+                    override suspend fun getSearchProjects(
+                        discoveryParams: DiscoveryParams,
+                        cursor: String?
+                    ): Result<SearchEnvelope> {
+                        params = discoveryParams
+                        return Result.success(SearchEnvelope(projectList))
+                    }
+                }).build()
+
+        viewModel = SearchAndFilterViewModel.Factory(environment).create(SearchAndFilterViewModel::class.java)
+
+        val searchState = mutableListOf<SearchUIState>()
+        backgroundScope.launch(dispatcher) {
+            viewModel.getPopularProjects()
+            viewModel.searchUIState.toList(searchState)
+        }
+
+        assertEquals(params?.sort(), DiscoveryParams.Sort.POPULAR)
```

Comment:
asserting with what params was `getSearchProjects` called, just `SORT` for now, will grow in following tasks